### PR TITLE
#CX-10274: Passing List Items Count To The Parent Fix

### PR DESCRIPTION
**Pull request by mithun-srinivas**
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/combobox/ComboBox.ts [hunk 1]
```diff
@@ -149,6 +150,16 @@ export namespace ComboBox {
     private multiSelected: number[] = [];
     private customContent: Element[] = [];
 
+    private notifylistitemscounttoparent() {
```
Comment (merajcal) [inline web-components/src/components/combobox/ComboBox.ts:153]: rename notifylistitemscounttoparent to notifySearchResultCount

File: web-components/src/components/combobox/ComboBox.ts [hunk 2]
```diff
@@ -149,6 +150,16 @@ export namespace ComboBox {
...
+      this.dispatchEvent(
+        new CustomEvent('item-count-changed', {
```
Comment (merajcal) [inline web-components/src/components/combobox/ComboBox.ts:155]: rename item-count-changed to md-combobox-search-result-count

**Review by merajcal — CHANGES_REQUESTED** [review]

**Review by vinitkumargoel — DISMISSED** [review]

**Comment by vinitkumargoel** [comment]
Please increase the version

File: web-components/src/components/combobox/ComboBox.ts (hunk 1 above)
Comment (mithun-srinivas) [reply web-components/src/components/combobox/ComboBox.ts]: Done

File: web-components/src/components/combobox/ComboBox.ts (hunk 2 above)
Comment (mithun-srinivas) [reply web-components/src/components/combobox/ComboBox.ts]: Done

**Comment by mithun-srinivas** [comment]
> Please increase the version

Done

File: web-components/src/components/combobox/ComboBox.ts [hunk 3]
```diff
@@ -149,6 +150,16 @@ export namespace ComboBox {
     private multiSelected: number[] = [];
     private customContent: Element[] = [];
 
+    private notifySearchResultCount() {
```
Comment (niti-owner) [inline web-components/src/components/combobox/ComboBox.ts:153]: Can you add function comments

File: web-components/src/components/combobox/ComboBox.ts [hunk 4]
```diff
@@ -1191,6 +1205,8 @@ export namespace ComboBox {
             bubbles: true
           })
         );
+      this.listItemsCount = this.filteredGroupOptions.length;
+      this.notifySearchResultCount()
```
Comment (niti-owner) [inline web-components/src/components/combobox/ComboBox.ts:1209]: 1. Can you end with ";" at the lines - 803 and 1209.
2. Can you add UTs for new code added.

File: web-components/src/components/combobox/ComboBox.ts (hunk 3 above)
Comment (mithun-srinivas) [reply web-components/src/components/combobox/ComboBox.ts:153]: Added comment.

File: web-components/src/components/combobox/ComboBox.ts (hunk 4 above)
Comment (mithun-srinivas) [reply web-components/src/components/combobox/ComboBox.ts]: Done

Comment (mithun-srinivas) [reply web-components/src/components/combobox/ComboBox.ts]: Screenshot 2024-03-11 at 3 30 40 PM

**Review by niti-owner — DISMISSED** [review]

File: web-components/src/components/combobox/ComboBox.ts [hunk 5]
```diff
@@ -149,6 +150,17 @@ export namespace ComboBox {
     private multiSelected: number[] = [];
     private customContent: Element[] = [];
 
+    private notifySearchResultCount() {
+      // this function is used to notify search result count.
+      this.dispatchEvent(
+        new CustomEvent('md-combobox-search-result-count', {
```
Comment (merajcal) [inline web-components/src/components/combobox/ComboBox.ts:156]: who will consume md-combobox-search-result-count event?

**Review by merajcal — DISMISSED** [review]

**Review by merajcal — DISMISSED** [review]

**Review by merajcal — APPROVED** [review]

File: web-components/src/components/combobox/ComboBox.stories.ts
Comment (iruchipriya) [inline web-components/src/components/combobox/ComboBox.stories.ts:1]: where are we using onSearchListUpdated function here?

**Review by iruchipriya — DISMISSED** [review]

**Review by iruchipriya — APPROVED** [review]

**Review by neelabhnagaich — APPROVED** [review]

**Review by merajcal — APPROVED** [review]

**Pull request merged**